Library/Formula/llvm.rb

```diff
@@ -111,6 +120,37 @@ class Llvm < Formula
   fails_with :gcc
   fails_with :llvm
 
+  # Disable superenv if building with additional runtime libraries (compiler-rt)
+  #
+  # Something about superenv confuses the linker when compiling the support
+  # libraries (compiler-rt). Linking fails with errors similar to:
+  #   ...
+  #   ld: building for iOS simulator, but linking against dylib built for OSX,
+  #     file '/usr/lib/libc++.dylib' for architecture x86_64
+  #   clang: error: linker command failed with exit code 1 (use -v to see invocation)
+  #   make[2]: *** [lib/clang/3.8.0/lib/darwin/
+  #     libclang_rt.ubsan_iossim_dynamic.dylib] Error 1
+  #   ...
+  #   fatal error: /Applications/Xcode-beta.app/Contents/Developer/Toolchains/
+  #     XcodeDefault.xctoolchain/usr/bin/lipo: specifed architecture type (i386)
+  #     for file (/tmp/llvm20150910-69149-1cdqr2s/./lib/clang/3.8.0/lib/darwin/
+  #     libclang_rt.builtins_i386_iossim.a) does not match its cputype
+  #     (16777223) and cpusubtype (3) (should be cputype (7) and cpusubtype (3))
+  #   make[2]: *** [projects/compiler-rt/lib/builtins/
+  #     libclang_rt.builtins_iossim.a] Error 1
+  #   <cleanup and display error>
+  # Unfortunately, the reason the linker is doing this is not known at this
+  # time (11 September 2015). This is not the only incident of its kind; see:
+  #     https://forums.developer.apple.com/thread/13795
+  # so it is possible that this can be removed if/when XCode and/or the CLT get
+  # fixed.
+  #
+  # Builds without these support libraries completed normally, which is why
+  # this change was not required earlier. In addition, the libraries are only
+  # built when clang is built
+
+  env :std if build.with? "clang"
```

Comment:
@DomT4 Got a piece of the puzzle. It seems setting `HOMEBREW_CCCFG` to anything including the `O` flag is causing only x86_64 architectures for part of the build to be built, instead of both x86_64 and i386. The commands printed in that section have `-arch x86_64 -arch i386` in them, so it wasn't the commands that were incorrect. Dropping into a shell in that section and prefixing `HOMEBREW_CCCFG=sag` to the compile command resulted in both architectures being built.

I'm not sure what the flag is normally; it's just `O` right now, but I think in previous runs it was `Osag`

Thoughts?